framework/commercejs/fetcher.ts

```diff
@@ -0,0 +1,61 @@
+import { commerce } from './lib/commercejs'
+import type { Fetcher } from '@commerce/utils/types'
+import { FetcherError } from '@commerce/utils/errors'
+
+// Fetches from an API route within /api/endpoints directory
+const customFetcher: Fetcher = async ({ method, url, body }) => {
+  const response = await fetch(url!, {
+    method,
+    body: body ? JSON.stringify(body) : undefined,
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  })
+    .then((response) => response.json())
+    .then((response) => response.data)
+
+  return response
+}
+
+const fetcher: Fetcher = async ({ url, query, method, variables, body }) => {
+  // If a URL is passed, it means that the fetch needs to be passed on to a custom API route.
+  const isCustomFetch = !!url
+  if (isCustomFetch) {
+    const data = await customFetcher({ url, method, body })
+    return data
+  }
+
+  // Fetch using the Commerce.js SDK, but make sure that it's a valid method.
+
+  const resource = commerce[query as keyof typeof commerce]
+  if (!resource) {
+    throw new FetcherError({
+      errors: [
+        { message: `Query ${query} does not exist on Commerce.js SDK.` },
+      ],
+      status: 400,
+    })
+  }
+
+  // @ts-ignore
```

Comment:
@goncy I've fixed the @ts-ignores and TODOs within the Commerce.js provider, except for the TODO regarding the checkout issue which you've since fixed. I'm going to work on that now, along with your code suggestions (thanks!)